src/main/java/Parser.java

```diff
@@ -0,0 +1,54 @@
+import java.time.DateTimeException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Parser {
+    public Parser() {
+
+    }
+    private static ArrayList<String> tokenize(String input) {
```

Comment:
Since you have been putting a newline between every function so far, might be good to keep consistent and put newline between functions here as well.